Add `CategoryBasis` for One-Hot Encoding of Discrete Variables (#527)
flatironinstitute/nemos: docs/how_to_guide/categorical_predictors/categorical_predictors.md

Reviewer: link to the technical note or have a dropdown computing the rank to show people how they can check this

> see [Identifiability of Categorical Predictors](categorical_identifiability).

Reviewer: ```suggestion
see [](categorical_identifiability).
```

:::

## Splitting a Continuous Variable by Category

> The primary use of the [`Category`](nemos.basis.Category) basis in NeMoS is to estimate category-specific tuning curves by multiplying it with a continuous basis.

Reviewer: ```suggestion
The [`Category`](nemos.basis.Category) basis in NeMoS also allows you to estimate category-specific tuning curves by multiplying it with a continuous basis.
```

feels weird to say "primary" here when it's the second thing on the page

> Continuing the previous example, let's assume that we have also recorded the average animal speed per trial and suppose we want to learn how the neuron responds to speed depending on the turn side. You can use the `Category` basis to produce an appropriate design matrix:

Reviewer: ```suggestion
Continuing the previous example, let's assume that we have also recorded the average animal speed per trial and suppose we want to learn how the neuron responds to speed depending on the turn side. You can multiply the `Category` basis by another basis to produce an appropriate design matrix:
```

```{code-cell} ipython3
speed = np.array([10., 3., 2., 20.])

# Category * continuous basis: one set of basis functions per category
bas = nmo.basis.Category(["L", "R"]) * nmo.basis.RaisedCosineLinearEval(3)
X = bas.compute_features(turn_side, speed)
print("X.shape: ", X.shape)  # (4, 6): 3 basis functions × 2 categories
```

> ## Complex Designs

Reviewer: ```suggestion

(complex-designs)=
## Complex Designs with `patsy` and `formulaic`
```

so we can link to this elsehwere

```{code-cell} ipython3
import pandas as pd
from patsy import dmatrix

> from formulaic import model_matrix

Reviewer: I think these should be just `import`, so it's clear when called later which package these are coming from (e.g., `patsy.dmatrix` instead of just `dmatrix`)

# patsy adds an intercept; drop it since NeMoS GLMs include one implicitly
design_df = design_df.drop(columns=["Intercept"])
print("patsy:\n\n", design_df)

# formulaic equivalent
design_df = model_matrix(formula, data)

> design_df = design_df.drop(columns=["Intercept"])

Reviewer: should add same note about dropping the intercept column here

> print("\n\nformulaic:\n\n", design_df)

Reviewer: still think this should be separate tabs

model = nmo.glm.GLM().fit(design_df, counts)
```

NeMoS [`Category`](nemos.basis.Category) basis provides a simple dummy coding of categorical variables. This is just one of the many encoding schemes that `patsy` provides.

> For example, the dummy code for one categorical predictor in NeMoS,

Reviewer: I would drop "dummy code" here and below. we don't explain what it means and it's not necessary here
